master/internal/stream/system.go

```diff
@@ -629,12 +668,14 @@ func (ss *SubscriptionSet) SubscriptionMod(ctx context.Context, msg Subscription
 	var msgs []interface{}
 	var err error
 	msgs, err = subMod(ctx, msgs, err, ss.Trials, add.Trials, drop.Trials)
+	msgs, err = subMod(ctx, msgs, err, ss.Metrics, add.Metrics, drop.Metrics)
```

Comment:
nit (non-blocking) comment: The only things that bugs me about this is that it's not immediately clear that msgs are being appended to, so it looks like msgs are being replaced.

Reply:
With the way it's written, it is replacing `msgs` in `SubscriptionMod`, but it's replaces it with a version that has the new messages appended to it, we could make it clearer by passing in a pointer to `msgs`.

Something like:
```
err := subMod(ctx, &msgs, err, ss.Trials,...)
err := subMod(ctx, &msgs, err, ss.Metrics,...)
```

I don't think it's really necessary to change for `subMod` but it might be worth while to consider for `startup`